src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectConfigurationStateSynchronizer.cs

```diff
@@ -28,9 +29,10 @@ internal class ProjectConfigurationStateSynchronizer : IProjectConfigurationFile
     private readonly ILogger _logger;
     private readonly TimeSpan _delay;
 
-    private readonly Dictionary<string, ProjectKey> _configurationToProjectMap;
     internal readonly Dictionary<ProjectKey, DelayedProjectInfo> ProjectInfoMap;
 
+    private ImmutableDictionary<string, ProjectKey> _filePathToProjectKeyMap = ImmutableDictionary<string, ProjectKey>.Empty;
```

Comment:
No comparer?

Reply:
oh, all paths are normalized on the way in?

Reply:
Correct. That's how it was implemented before. I'll take a closer and see if a comparer would just be simpler.

Reply:
I took a look, and I was wrong -- it *did* use a comparer before my change. That might actually be redundant, but I'm going to add the comparer since it was an accident to not include it.